### PR TITLE
sched/environ: Improve performance of set/unset environment

### PR DESCRIPTION
**Pull request by anchao**
## Summary

sched/environ: Improve performance of set/unset environment

reserve dimensional array of environment to avoid frequent scaling

Signed-off-by: chao an <anchao@lixiang.com>

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

File: sched/environ/env_removevar.c [hunk 1]
```diff
@@ -86,15 +86,18 @@ void env_removevar(FAR struct task_group_s *group, ssize_t index)
...
     {
       /* Reallocate the environment to reclaim a little memory */
 
       group->tg_envp = group_realloc(group, group->tg_envp,
-                                     sizeof(*group->tg_envp) *
-                                     (group->tg_envc + 1));
+         sizeof(*group->tg_envp) * (group->tg_envc +
+                                    SCHED_ENVIRON_RESERVED + 1));
       DEBUGASSERT(group->tg_envp != NULL);
+      group->tg_envpc = group->tg_envc + SCHED_ENVIRON_RESERVED + 1;
```
Comment (xiaoxiang781216) [inline sched/environ/env_removevar.c:100]: move before line 94

File: sched/environ/env_removevar.c [hunk 2]
```diff
@@ -86,15 +86,18 @@ void env_removevar(FAR struct task_group_s *group, ssize_t index)
...
+  else if (group->tg_envc <=
+           (group->tg_envpc - SCHED_ENVIRON_RESERVED * 2))
     {
       /* Reallocate the environment to reclaim a little memory */
 
       group->tg_envp = group_realloc(group, group->tg_envp,
-                                     sizeof(*group->tg_envp) *
-                                     (group->tg_envc + 1));
+         sizeof(*group->tg_envp) * (group->tg_envc +
+                                    SCHED_ENVIRON_RESERVED + 1));
```
Comment (xiaoxiang781216) [inline sched/environ/env_removevar.c:98]: group->tg_envpc

File: sched/environ/env_removevar.c (hunk 1 above)
Comment (anchao) [reply sched/environ/env_removevar.c]: Done

File: sched/environ/env_removevar.c (hunk 2 above)
Comment (anchao) [reply sched/environ/env_removevar.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**